scripts/clear-signed-agreements.py

```diff
@@ -46,22 +48,41 @@ def main(stage, framework_slug, api_token, user):
                     extra={'supplier_id': supplier['supplierId']})
         api_client.unset_framework_agreement_returned(supplier['supplierId'], framework_slug, user)
 
-    signed_agreements = filter(
-        lambda x: re.search(r'/(\d+)-signed-framework-agreement.pdf', x['path']),
-        agreements_bucket.list('{}/agreements/'.format(framework_slug))
-    )
+    signed_agreements = filter_agreements(supplier_ids)
 
     for document in signed_agreements:
         logger.info("Deleting {path}", extra={'path': document['path']})
         agreements_bucket.delete_key(document['path'])
 
 
+def filter_agreements(supplier_ids=None):
+    if supplier_ids:
+        matcher = lambda x: x['path'] in ['{}-signed-framework-agreement.pdf'.format(id) for id in supplier_ids]
+    else:
+        lambda x: re.search(r'/(\d+)-signed-framework-agreement.pdf', x['path'])
+
+    return filter(
+        matcher,
+        agreements_bucket.list('{}/agreements/'.format(framework_slug))
+    )
+
+
+def get_supplier_ids_from_csv(csv_path):
+    with open(csv_path, 'r') as csvfile:
+        for row in csv.reader(csvfile):
+            # csv should only have one row
+            return row
+
+
 if __name__ == '__main__':
     arguments = docopt(__doc__)
     stage = arguments['<stage>']
     framework_slug = arguments['<framework_slug>']
     data_api_access_token = arguments['--api-token']
+    supplier_ids = None
 
     user = getpass.getuser()
+    if arguments['--csv_path']:
```

Comment:
If it's just a single row of supplier IDs they could just be passed in on the command line rather than in a file?  Not sure... maybe that's more prone to user error?